src/app/services/extras/runge-kutta.ts

```diff
@@ -99,14 +100,20 @@ export class RungeKutta {
 
     let result: { pos: Vector3, dir: Vector3 }[] = [];
 
-    while (rkState.pathLength < plength) {
+    while (rkState.pathLength < plength ) {
       rkState.pathLength += RungeKutta.step(rkState);
       // Cloning state to avoid using the reference
       let copiedState = JSON.parse(JSON.stringify(rkState));
       result.push({
         pos: copiedState.pos,
         dir: copiedState.dir
       });
+      // Ugly
+      if ( inbounds(copiedState.pos) != true){
```

Comment:
If we have the parameter above typed correctly (so the compiler knows that it must return boolean), I think it's safe to use just the function - `if (inbounds(copiedState.pos))`.